custom/icds/models.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.functional import cached_property
+from django.utils.translation import ugettext_lazy as _
+
+from corehq.apps.app_manager.dbaccessors import get_build_by_version
+from corehq.motech.utils import b64_aes_decrypt
+from custom.icds.utils.ccz_hosting import CCZHostingUtility
+from custom.icds.validators import (
+    LowercaseAlphanumbericValidator,
+)
+
+
+class CCZHostingLink(models.Model):
+    identifier = models.CharField(null=False, unique=True, max_length=255, db_index=True,
+                                  validators=[LowercaseAlphanumbericValidator])
+    username = models.CharField(null=False, max_length=255)
+    # b64_aes_encrypt'ed raw password is stored in DB
+    password = models.CharField(null=False, max_length=255)
+    domain = models.CharField(null=False, max_length=255)
+    page_title = models.CharField(blank=True, max_length=255)
+
+    def __str__(self):
+        return self.identifier
+
+    @cached_property
+    def get_password(self):
+        return b64_aes_decrypt(self.password)
+
+    def to_json(self):
+        from custom.icds.serializers import CCZHostingLinkSerializer
+        return CCZHostingLinkSerializer(self).data
+
+    def delete(self, *args, **kwargs):
+        for ccz_hosting in CCZHosting.objects.filter(link=self):
+            ccz_hosting.delete()
+        super(CCZHostingLink, self).delete(*args, **kwargs)
+
+
+class CCZHosting(models.Model):
```

Comment:
i wanted it to be different from any vocabulary used already to avoid confusion so just `CCZ` would beat that.
`UploadedCCZ` does seem right because the CCZ is not being uploaded (anymore :), thats the aim of this PR ).
CCZs are being hosted for use, hence CCZHosting.
Makes sense?
(tbh, i personally don't love that name but could not come up with anything better)

Reply:
Maybe `HostedCCZ`?  I try to think of model names as a simple descriptor of one row in the database - as in "this user created a new X".  I guess `CCZHosting` works in that case too.  Whatever you prefer.

Reply:
Agree `HostedCCZ` would be better, if only because it's more obviously a noun.

Reply:
renamed this particular model [here](https://github.com/dimagi/commcare-hq/pull/24129/commits/6782754a1139e3e832a2d12043cbf44f1bf3acac)

Left other models/views as-is. Hopefully this wont be confusing now since model is HostedCCZ and other models/views are like `CCZHostingLink`, `CCZHostingSupportingFile`, `ManageCCZHosting` or probably i just got accustomed to everything starting with `CCZHosting`

Reply:
Would be ideal to change all instances of `CCZHosting` to `HostedCCZ` for consistency.

- `CCZHostingSupportingFile` -> `HostedCCZSupportingFile`
- `ManageCCZHosting` -> `ManageHostedCCZ`

Reply:
okay. Updated in [commit](https://github.com/dimagi/commcare-hq/pull/24129/commits/23ce553c113c4aa36e5e239cf11283ff519f937f). Only the toggle says CCZ Hosting which seemed fine.